fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -42,52 +44,113 @@ impl<T> Interpreter<PredicateStorage, T> {
     /// This is not a valid entrypoint for debug calls. It will only return a `bool`, and not the
     /// VM state required to trace the execution steps.
     pub fn check_predicates<Tx>(
-        checked: Checked<Tx>,
+        checked: &Checked<Tx>,
         params: ConsensusParameters,
         gas_costs: GasCosts,
     ) -> Result<PredicatesChecked, PredicateVerificationFailed>
     where
         Tx: ExecutableTransaction,
-        <Tx as IntoChecked>::Metadata: CheckedMetadata,
+        <Tx as IntoChecked>::CheckedMetadata: CheckedMetadata,
     {
         if !checked.transaction().check_predicate_owners(&params) {
             return Err(PredicateVerificationFailed::InvalidOwner);
         }
 
         let mut vm = Interpreter::with_storage(PredicateStorage::default(), params, gas_costs);
 
+        let mut cumulative_gas_used: Word = 0;
+
+        vm.init_predicate(checked.clone());
```

Comment:
Was there a reason to move this before the following line?

Reply:
can you clarify what you are referencing?

Reply:
It was moved above another line but that line has since been removed.